CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/ImportDelimitedPlugin.java

```diff
@@ -384,31 +386,40 @@ private static Map<String, Integer> processVertices(final ImportDefinition defin
             interaction.setProgress(++currentRow, totalRows, "Importing Vertices: " + source, true);
 
             final String[] row = data.get(i);
-            if (filter == null || filter.passesFilter(i - 1, row)) {
+            
+            try {
+                if (filter == null || filter.passesFilter(i - 1, row)) {
 
-                final int vertexId = graph.addVertex();
+                    final int vertexId = graph.addVertex();
 
-                try {
+                    try {
 
-                    for (final ImportAttributeDefinition attributeDefinition : attributeDefinitions) {
-                        attributeDefinition.setValue(graph, vertexId, row, (i - 1));
-                    }
+                        for (final ImportAttributeDefinition attributeDefinition : attributeDefinitions) {
+                            attributeDefinition.setValue(graph, vertexId, row, (i - 1));
+                        }
 
-                    if (initialiseWithSchema && graph.getSchema() != null) {
-                        graph.getSchema().completeVertex(graph, vertexId);
-                    }
+                        if (initialiseWithSchema && graph.getSchema() != null) {
+                            graph.getSchema().completeVertex(graph, vertexId);
+                        }
 
-                    // Count the number of processed rows to notify in the status message
-                    ++importedRows;
+                        // Count the number of processed rows to notify in the status message
+                        ++importedRows;
 
-                } catch (final DateTimeException | IllegalArgumentException | SecurityException ex) {
-                    if (skipInvalidRows) {
-                        graph.removeVertex(vertexId);
-                        ++skippedRow;
-                    } else {
-                        throw ex;
+                    } catch (final DateTimeException | IllegalArgumentException | SecurityException ex) {
+                        if (skipInvalidRows) {
+                            graph.removeVertex(vertexId);
+                            ++skippedRow;
+                        } else {                            
+                            throw ex;
+                        }
                     }
                 }
+            } catch (final DateTimeException | IllegalArgumentException | SecurityException ex) {
+                if (skipInvalidRows) {          
+                    ++skippedRow;
+                } else {                    
+                    throw ex;
+                }
```

Comment:
There are nested try-catch blocks with exactly the same catch conditions.
This can be re-written to only use the outer try-catch block. We can move the `graph.removeVertex(vertexId);` to the outer catch block, but make it conditional on whether there was a new vertex added.
ie. initialise vertexId to -1 before the if statement, then in the catch block,  if vertexId != -1, remove the vertex.